packages/devextreme/js/__internal/ui/chat/messagebox.ts

```diff
@@ -36,20 +44,30 @@ class MessageBox extends DOMComponent<MessageBox, Properties> {
 
   _messageSendAction?: (e: Partial<MessageSendEvent>) => void;
 
+  _typingStartAction?: (e: Partial<TypingStartEvent>) => void;
+
+  _typingEndAction?: () => void;
+
+  _typingEndTimeoutId?: ReturnType<typeof setTimeout> | undefined;
```

Comment:
This argument is optional. Does it require a `| undefined` declaration?

Reply:
Yes, because in the _clearTypingEndTimeout method and in _updateTypingEndTimeout I set _typingEndTimeoutId = undefined. This type does not match the ReturnType<typeof setTimeout>. Even though we specified that the property is optional, TS considers it an error if we do not explicitly specify the type | undefined